rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java

```diff
@@ -664,7 +680,77 @@ protected Integer commitFederation(boolean dryRun, Keccak256 hash, BridgeEventLo
         logger.debug("[commitFederation] New Federation committed: {}", newFederation.getAddress());
         eventLogger.logCommitFederation(rskExecutionBlock, oldFederation, newFederation);
 
-        return FederationChangeResponseCode.SUCCESSFUL.getCode();
+        return FederationChangeResponseCode.SUCCESSFUL;
+    }
+
+    /**
+     * Commits the currently pending federation to be the proposed one.
+     * That is, the proposed federation is set to be a new federation generated from the currently pending federation,
+     * and the pending federation is wiped out.
+     * @param dryRun whether to just do a dry run
+     * @param hash the pending federation's hash. This is checked to match the execution block's pending federation hash.
+     * @return SUCCESSFUL upon success, PENDING_FEDERATION_NON_EXISTENT if there was no pending federation,
+     * INSUFFICIENT_MEMBERS if the pending federation was incomplete,
+     * PENDING_FEDERATION_MISMATCHED_HASH if the given hash doesn't match the current pending federation's hash.
+     */
+    private FederationChangeResponseCode commitFederation(boolean dryRun, Keccak256 hash, BridgeEventLogger eventLogger) {
```

Comment:
Seems to be a lot of code repeated between both `commitFederation` methods. Let's create a task to analyze re-using this code

Reply:
sure

Reply:
already addressed in this pr